### PR TITLE
Refine test workflows and integrate uncommitted changes check

### PR DESCRIPTION
**Pull request by clobrano**
- Update test target in Makefile to run unit tests followed by
  uncommitted changes verification.
- Enhance `verify-diff.sh` to detect uncommitted changes
  and out-of-date bundle.
- Remove mutation testing references from CI and Makefile.

### CODE REVIEW

File: hack/verify-diff.sh [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
 fi
 
-git diff --exit-code
+# Checks that changes under config directory are reflected in the bundle directory.
+# This is to ensure that the bundle is up-to-date with the latest changes.
+# If this fails, run 'make bundle' to update the bundle.
+if [[ -n "$(git status --porcelain config/)" ]] && [[ -z "$(git status --porcelain bundle/)"  ]]; then
+    echo ""
+    echo "[!] Config folder has changes, but Bundle does not. Check if Bundle is out-of-date."
+    echo "$(git status --porcelain config/)"
```
Comment (razo7) [inline hack/verify-diff.sh:17]: Maybe also printing with  `echo "$(git status --porcelain bundle/)"`

Comment (clobrano) [reply hack/verify-diff.sh:17]: I don't understand, in this case bundle folder is not expected to have any change to show (unless the check above is wrong somehow)

Comment (razo7) [reply hack/verify-diff.sh:17]: You are right, I was confused and I didn't see the `-z` beforehand..

File: hack/verify-diff.sh [hunk 2]
```diff
@@ -1,10 +1,22 @@
 #!/bin/bash
-FILE_DIFF=$(git ls-files -o --exclude-standard)
 
-if [ "$FILE_DIFF" != "" ]; then
-  echo "Found untracked files:"
-  echo "$FILE_DIFF"
-  exit 1
+status=0
+# Checks that there are no uncommitted changes.
+if [[ -n "$(git status --untracked-files=no --porcelain .)" ]]; then
```
Comment (slintes) [inline hack/verify-diff.sh:5]: please remove  `-untracked-files=no` and find out why you don't see the added files locally (I see them after `make go-tidy go-vendor`)

/hold

**Review by slintes — CHANGES_REQUESTED** [review]

File: hack/verify-diff.sh (hunk 2 above)
Comment (slintes) [reply hack/verify-diff.sh:5]: also, please put such workarounds in a commit on its own for better visibilty

**Review by slintes — APPROVED** [review]
/hold cancel

**Pull request merged**